pkg/controller/daemon/daemon_controller.go

```diff
@@ -774,9 +774,14 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *apps.DaemonSet) (map[s
 	// Group Pods by Node name.
 	nodeToDaemonPods := make(map[string][]*v1.Pod)
 	for _, pod := range claimedPods {
-		nodeName := pod.Spec.NodeName
+		nodeName, err := util.GetTargetNodeName(pod)
```

Comment:
Here it seems to be changing the behavior if the feature is disabled by returning error if pod.Spec.NodeName is 0.